lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -577,10 +577,13 @@ def _set_credentials(self):
 
         if not self.boto_profile and not (self.aws_access_key_id and self.aws_secret_access_key):
             session = botocore.session.get_session()
-            if session.get_credentials() is not None:
-                self.aws_access_key_id = session.get_credentials().access_key
-                self.aws_secret_access_key = session.get_credentials().secret_key
-                self.aws_security_token = session.get_credentials().token
+            credentials = session.get_credentials()
```

Comment:
I think it may be cleaner (and less nested) to use EAFP style:
```suggestion
			try:
            	credentials = session.get_credentials().get_frozen_credentials()
            except AttributeError:
            	pass
            else:
            	self.aws_access_key_id = credentials.access_key
                self.aws_secret_access_key = credentials.secret_key
                self.aws_security_token = credentials.token

```